protocol/contracts/leaser/src/state/config.rs

```diff
@@ -61,21 +62,25 @@ impl Config {
         lease_due_period: Duration,
     ) -> ContractResult<()> {
         Self::STORAGE
-            .update(storage, |mut c| -> ContractResult<Config> {
-                c.lease_interest_rate_margin = lease_interest_rate_margin;
-                c.lease_position_spec = lease_position_spec;
-                c.lease_due_period = lease_due_period;
-                Ok(c)
+            .update(storage, |c| -> ContractResult<Config> {
+                Ok(Self {
+                    lease_interest_rate_margin,
+                    lease_position_spec,
+                    lease_due_period,
+                    ..c
+                })
             })
             .map(|_| ())
```

Comment:
I didn't write it for every single instance, but if you'd be accepting of using `drop`, it might be good to be consistent.

Reply:
hahaha
expanding the scope of this PR may go endless

Reply:
True, yet it was just a suggestion after all. We could've just left it as a future work. I didn't really know how much it would extend the scope of the PR.